### ISSUE TITLE
Regression -- Cannot run one-off processes with --release specified on processes with --count=-1

### ISSUE THREAD

**Issue by snario** [convox/rack#838]
## Setup
```bash
$ convox scale -a <app>
NAME       DESIRED  RUNNING  CPU  MEMORY
<process>  -1        0        0    256
```

## Before
```bash
$ convox run <process> <command> -a <app> --release <release_id>
... runs the command ...
```
## After
```bash
$ convox run <process> <command> -a <app> --release <release_id>
ERROR: unable to find contaier for frontend
```

## Release that broke this behaviour
https://github.com/convox/rack/releases/tag/20160629165916

**Comment by MiguelMoll** [comment]
@snario Thanks for bringing this to our attention. I was able to reproduce this issue only with a release that hasn't been promoted. Is that the same case with you?

```
⟩ convox scale
NAME  DESIRED  RUNNING  CPU  MEMORY
web   -1       0        0    256

⟩ convox run web 'echo $RELEASE' -a httpd --release RBTHOWKEUPR
RBTHOWKEUPR

⟩ convox releases
ID           CREATED              BUILD        STATUS
RKPLJKNKDTQ  38 seconds from now  BITGKJEQOUU
RBTHOWKEUPR  22 minutes ago       BITGKJEQOUU  active
RVQNQAUQHUH  1 week ago           BMTUZLBHMTQ

⟩ convox run web 'echo $RELEASE' -a httpd --release RKPLJKNKDTQ
ERROR: unable to find contaier for web
```

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;bFCGG3GT&#x2F;313-fix-convox-run-work-with-a-unpromoted-release"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Fix convox run work with a &quot;unpromoted&quot; release</a>

**Comment by snario** [comment]
@MiguelMoll you are right it is for un-promoted releases, which in my circumstances (count=-1) is meaningless since promoting a release really has no effect. Thank you though for pointing that out as it is a workaround (although an unfortunate additional step for me when deploying now)

**Comment by mwarkentin** [comment]
@snario You can use `convox deploy` to build and promote in one step. If it's an environment variable you're setting you can do that in one step too with `convox env set <key>=<val> --promote`

**Comment by snario** [comment]
@mwarkentin I use the GitHub integration and on some deploys need to run a one-off process, so deploying looks like:

```bash
# 1. Merge a PR to master
# 2. Wait for the build to finish ...
convox run frontend webpack --id=$xyz
# 3. Wait for this to finish 
convox env set FRONTEND_BUILD_ID=$xyz --promote
```

**Comment by mwarkentin** [comment]
Ah, gotcha..

**Comment by MiguelMoll** [comment]
The fix for this went out in #891.
